packages/components/src/components/files/FileAttachmentContainer.tsx

```diff
@@ -250,9 +251,9 @@ export class FileAttachmentContainer extends React.Component<FileAttachmentConta
                     <input
                         accept={acceptedFormats}
                         className="file-upload--input"
-                        id="fileUpload"
+                        id={"fileUpload" + this.props.index}
```

Comment:
I think you might want that same code here: `(index !== undefined ? index : '')`